azure-pipelines/templates/auto-doc-job.yml

```diff
@@ -0,0 +1,54 @@
+# File build-test-job.yml
```

Comment:
I think i would combine the two yml files.  There is no reason you need to use templates.  You are noting trying to support numerous environments or toplevel files, right?  We do this in other places because we want to reuse the same steps for linux and windows and OSX(if we did this).  that would never be the case for a release like this.